com.unity.ml-agents/Runtime/Analytics/Events.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Collections.Generic;
+using Unity.MLAgents.Actuators;
+using Unity.MLAgents.Sensors;
+
+namespace Unity.MLAgents.Analytics
+{
+    internal struct InferenceEvent
+    {
+        public string BehaviorName;
+        public string BarracudaModelSource;
+        public string BarracudaModelVersion;
+        public string BarracudaModelProducer;
+        public string BarracudaPackageVersion;
+        public int InferenceDevice;
+        public List<EventObservationSpec> ObservationSpecs;
+        public EventActionSpec ActionSpec;
+        public int MemorySize;
+        // public Int64 ModelHash; //TODO ?
+    }
+
+    /// <summary>
+    /// Simplified version of ActionSpec struct for use in analytics
+    /// </summary>
+    internal struct EventActionSpec
+    {
+        public int NumContinuousActions;
+        public int NumDiscreteActions;
+        public int[] BranchSizes;
+
+        public static EventActionSpec FromActionSpec(ActionSpec actionSpec)
+        {
+            var branchSizes = actionSpec.BranchSizes ?? Array.Empty<int>();
+            return new EventActionSpec
+            {
+                NumContinuousActions = actionSpec.NumContinuousActions,
+                NumDiscreteActions = actionSpec.NumDiscreteActions,
+                BranchSizes = branchSizes,
+            };
+        }
+    }
+
+    /// <summary>
+    /// Simplified summary of Agent observations for use in analytics
+    /// </summary>
+    internal struct EventObservationSpec
+    {
+        public string SensorName;
+        public int[] ObservationShape;
```

Comment:
How future proof is this? Will we be able to add fields to this later (when we add dimension properties or types to observation specs?

Reply:
We can version the event schema and add fields, although I don't think it will be automatic. So if we know there are things to add, we should do it now. Guess we should add a field for compression, and maybe a placeholder for the properties flags.

Reply:
There could be a field which is a array. Either key/value or just a key.
Then the key/value or key can be added in you code and the data will be sent